packages/@react-stately/virtualizer/src/Virtualizer.ts

```diff
@@ -431,6 +431,9 @@ export class Virtualizer<T extends object, V, W> {
     if (this._relayoutRaf) {
       cancelAnimationFrame(this._relayoutRaf);
       this._relayoutRaf = null;
+      // tentative: Update the provided context with the current invalidationContext since we are cancelling
+      // a scheduled relayoutNow call that has this._invalidationContext as its context arg (aka the relayoutNow in relayout)
+      context = {...this._invalidationContext, ...context};
```

Comment:
Tentative fix, noticed that the cases where the combobox items were sized incorrectly were because the following [`relayoutNow`](https://github.com/adobe/react-spectrum/blob/main/packages/@react-stately/virtualizer/src/Virtualizer.ts#L419-L422) wouldn't happen in time and would get cancelled by a [relayoutNow call from a transaction](https://github.com/adobe/react-spectrum/blob/main/packages/@react-stately/virtualizer/src/Virtualizer.ts#L1088). This would result in the cached item size being used since the invalidationContext isn't provided in the transaction call. Still not 100% sure if I can do something further downstream instead (haven't had any luck messing around in useVirtualizerItem)  

Alternative fixes:
- provide a default width (`192` or something) to menuWidth in [ComboBox](https://github.com/adobe/react-spectrum/blob/main/packages/@react-spectrum/combobox/src/ComboBox.tsx#L120).
- Use the original approach mentioned in the [issue](https://github.com/adobe/react-spectrum/issues/1485#issuecomment-767386501)

Alternative paths to explore still:
- Picker's layoutInfo width goes from 0 to its final calculated width of 190px and thus doesn't have the same issue. ComboBox's layout width goes from 0 to 30 to 188 for some reason and runs into this issue. Figure out why

Reply:
Turns out Picker is different from ComboBox because the collection from ComboBox is changed from `collection` to `filteredCollection` which then doesn't pass the equality check in [setData](https://github.com/adobe/react-spectrum/blob/main/packages/@react-stately/virtualizer/src/Virtualizer.ts#L204) even if the contents are the same. This causes the transaction to happen which cancels the relayoutNow call mentioned previously.